gensim/models/doc2vec.py

```diff
@@ -906,6 +907,12 @@ def infer_vector(self, doc_words, alpha=None, min_alpha=None, epochs=None, steps
             The inferred paragraph vector for the new document.
 
         """
+        if isinstance(doc_words, string_types):
+            warnings.warn(
+                "The parameter `doc_words` takes a list of strings, while you passed a single string, "
+                "which will be accepted and treated as a list of characters. Is this what you wanted?"
```

Comment:
It's never what users wanted. I'd prefer an exception here.

And if users really want to pass in a list of characters, they can pass in a list of characters (`list(string)`, not a string).